doc/tutorial/search/search_solution5.cpp

```diff
@@ -0,0 +1,93 @@
+#include <seqan3/alignment/configuration/all.hpp>
+#include <seqan3/alignment/pairwise/align_pairwise.hpp>
+#include <seqan3/io/stream/debug_stream.hpp>
+#include <seqan3/search/algorithm/search.hpp>
+
+using namespace seqan3;
+
+void run_text_single()
+{
+    dna4_vector text{"CGCTGTCTGAAGGATGAGTGTCAGCCAGTGTAACCCGATGAGCTACCCAGTAGTCGAACTGGGCCAGACAACCCGGCGCTAATGCACTCA"_dna4};
+    dna4_vector query{"GCT"_dna4};
+    fm_index index{text};
+
+    using it_t = decltype(std::ranges::begin(text));
+
+    debug_stream << "Searching all best hits allowing for 1 error in a single text\n";
+
+    configuration const search_config = search_cfg::max_error{search_cfg::total{1}} |
+                                        search_cfg::mode{search_cfg::all_best};
+    configuration const align_config = align_cfg::edit |
+                                       align_cfg::aligned_ends{align_cfg::seq1_ends_free} |
+                                       align_cfg::result{align_cfg::with_trace};
+
+    auto results = search(index, query, search_config);
+
+    debug_stream << "There are " << results.size() << " hits.\n";
+    debug_stream << "-----------------\n";
+
+    for (auto && pos : results)
+    {
+        size_t start = pos ? pos - 1 : 0;
+        size_t end   = pos + query.size() + 1;
+        auto text_view = view::subrange<it_t, it_t>{std::ranges::begin(text) + start,
+                                                    std::ranges::begin(text) + end};
```

Comment:
you could write instead:
```cpp
std::span text_view{std::data(text) + start, query.size() + 1};
```

And then get rid of the typedef and `size_t end`

Reply:
Almost works. Works if I just want to print it. For the alignment we have the problem that we need to be able to call `size()` on it.

In the current span.hpp `size` is a data member and the member function is `get_size()`

Reply:
what, why?  :rage2: 

it should just be `.size()` 
https://en.cppreference.com/w/cpp/container/span

Reply:
I will do a search and replace on it

Reply:
Yes, it should be `.size()`.

When I do search/replace (I just renamed the `size` to `_size`), I get static asserts `static_assert(idx >= 0 && idx < get_size(), "span<T>[] index out of bounds");` for the alignment module.

If you manage to make it work, I can change it 😁

Reply:
I have made a PR. I also found your issue and fixed it hopefully.